### PR TITLE
Add CI job for Go 1.18 running on Ubuntu 22.04 (jammy)

### PR DESCRIPTION
**Pull request by jayofdoom**
This allows us to test both Ubuntu LTS golang versions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -175,6 +175,13 @@ jobs:
       - store_test_results:
           path: test_reports/junit.xml
 
+  checks_and_unit_tests-ubuntu2204-go118:
+    <<: *checksAndUnitTests
+    machine:
+      docker_layer_caching: true
+      image: ubuntu-2204:current
```
Comment (severinson) [inline .circleci/config.yml:182]: I'd suggest using a specific version to avoid it changing behind our back.

File: .circleci/config.yml [hunk 2]
```diff
@@ -175,6 +175,13 @@ jobs:
       - store_test_results:
           path: test_reports/junit.xml
 
+  checks_and_unit_tests-ubuntu2204-go118:
+    <<: *checksAndUnitTests
```
Comment (severinson) [inline .circleci/config.yml:179]: I guess this is some code reuse feature of circleci?

File: .circleci/config.yml [hunk 3]
```diff
@@ -175,6 +175,13 @@ jobs:
...
+    machine:
+      docker_layer_caching: true
+      image: ubuntu-2204:current
+      resource_class: xlarge
+
   build_and_integration_tests:
     machine:
```
[the remark this answers is not in the document]
Comment (jayofdoom) [reply .circleci/config.yml:186]: This is the integration test portion I did not modify. It has not changed and has a full YAML map of settings; albeit most of them are hidden by default in github.

**Comment by severinson** [comment]
I think this looks good and I'd be happy to merge it. I'll chat with James (who's an admin on the repo) to merge and update the master branch protection settings accordingly.

**Review by JamesMurkin — APPROVED** [review]

**Pull request merged**